### PR TITLE
Feature/inputs

### LINKED ISSUE #40: Finalize input edit/cancel on click functionality
https://www.youtube.com/watch?v=WTh54FMNrbU

### PR DESCRIPTION
**Pull request by tashiad**
### Description
* Makes `My Account` into a class component with a controlled form
* Description, email, and address fields display current state as a default, then turn into inline input fields when you click the edit button. When you update the input and click the check box, state is updated and the input field goes away (back to the default state with the new changes reflected).
* Sets up my environment after issues on Rachel's branch were resolved (so this was branched off of Rachel's `Feature/my-account` branch since hers hasn't been merged yet)

#### Closes Issues 
closes #40 

## Proposed Changes
`MyAccount.js`
  
## Screenshots if applicable (Before/After)

* We'll need to hook the state changes up to Redux/the API
* Seems like this could be refactored... at least I think email and address could be. Open to thoughts/another pair of eyes!
* Styling isn't 100% there but figured it's good enough for desktop MVP
* I left image and delivery date alone as these will require a different approach

### CODE REVIEW

File: code/web/src/modules/user/MyAccount.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
+//Imports
+import React, { PureComponent } from 'react'
```
Comment (srslie) [inline code/web/src/modules/user/MyAccount.js:2]: Still a bit iffy on PureComponents vs normal components, and I know it vaguely has something to do with a built-in shallow reload, but couldn't speak to it well. I know we're just mirroring their code here but if someone understands it more I'd love to hear an explanation for one over the other.

File: code/web/src/modules/user/MyAccount.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      description: 'Fashion lover!',
+      email: 'user@crate.com',
+      address: '7575 E 57th Dr, Denver, CO, 12543',
+      descriptionEditMode: false,
+      emailEditMode: false,
+      addressEditMode: false
+    }
+  }
+
+  handleChange = event => {
```
Comment (srslie) [inline code/web/src/modules/user/MyAccount.js:35]: Nice reusability here!

File: code/web/src/modules/user/MyAccount.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+                {/* Profile Image */}
+                <div style={{ width: '200px', height: '200px', position: 'relative', overflow: 'hidden', borderRadius: '50%' }}>
+                  <img
+                    src='https://hips.hearstapps.com/countryliving.cdnds.net/17/47/2048x1365/gallery-1511194376-cavachon-puppy-christmas.jpg?resize=768:*'
+                    alt='profile image'
+                    style={{ display: 'inline', margin: '0 auto', marginLeft: '-25%', height: '100%', width: 'auto' }}
+                  />
+                </div>
+                <File
```
Comment (srslie) [inline code/web/src/modules/user/MyAccount.js:199]: If we wanted to match this to the button styling of the other 'edit' buttons here, we could probably pass an inline style that would overwrite the component one, but I think it looks nice either way.

File: code/web/src/modules/user/MyAccount.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                  type='file'
+                  name='image'
+                  text='Upload Image'
+                  value={this.state.image}
+                  onChange={this.handleChange}
+                />
+
+                {/* Personal Description */}
+                <p style={{ color: grey2 }}><strong>Personal Description:</strong></p>
```
Comment (srslie) [inline code/web/src/modules/user/MyAccount.js:208]: A minor css edit might bring the label down a touch to be closer to the description than the image upload to be more sectional.

File: code/web/src/modules/user/MyProducts.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      viewAll: true
+    }
+  }
+
+  toggleKept = () => {
+    // event.preventDefault()
```
Comment (srslie) [inline code/web/src/modules/user/MyProducts.js:31]: If we're not passing in event then we can lose that line or refactor to use it, seems to function okay.

File: code/web/src/modules/user/MyProducts.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                  <tr>
+                    <th>Image</th>
+                    <th>Name</th>
+                    <th>Description</th>
+                    <th>Delivered On</th>
+                    <th style={{ textAlign: 'center' }} >Kept</th>
+                  </tr>
+                </thead>
+
+                {/*This is where we'll map through all state user products and return the tr:*/}
```
Comment (srslie) [inline code/web/src/modules/user/MyProducts.js:72]: Thanks for putting this in as a comment, I should've done this as well as the hardcoding of the example rows. This seems solid to me and I'll add the logic of the toggle to this!

**Review by srslie — APPROVED** [review]

File: code/config/config.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "development": {
-    "username": "isabellevillasenor",
+    "username": "rachelbuchta",
```
Comment (gaj23) [inline code/config/config.json:3]: I would like to leave a comment inside of the README about this needed change & need to download sequelize.

File: code/web/src/modules/user/MyAccount.js (hunk 1 above)
Comment (gaj23) [reply code/web/src/modules/user/MyAccount.js:2]: My understanding is that it helps with performance in some cases and doesn't reload as much or often and is best used for small UI POV. [helpful & free article](https://medium.com/groww-engineering/stateless-component-vs-pure-component-d2af88a1200b)

File: code/web/src/modules/user/MyProducts.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  toggleKept = () => {
+    // event.preventDefault()
+    this.setState({ viewAll: !this.state.viewAll })
+  }
+
+  render() {
+    const kept = true
+    const kept2 = false
```
Comment (gaj23) [inline code/web/src/modules/user/MyProducts.js:37]: For readability, I would like to pull the kept into it's own function that returns a boolean value based on the data we receive.

File: code/web/src/modules/user/MyAccount.js [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+    event.preventDefault()
+
+    this.setState({
+      descriptionEditMode: !this.state.descriptionEditMode
+    })
+  }
+
+  renderDescriptionEditView = () => {
+    return (
+      <div style={{ display: 'flex', alignItems: 'center' }}>
```
Comment (gaj23) [inline code/web/src/modules/user/MyAccount.js:53]: This is really cool to see in action and feels good from the UX/UI pov.

File: code/web/src/modules/user/MyProducts.js (hunk 6 above)
Comment (gaj23) [reply code/web/src/modules/user/MyProducts.js:72]: Rows are going to be great for testing purposes too!

**Review by gaj23 — APPROVED** [review]
Thank you for your additions to the `My Accounts` page! I'm looking forward to hooking up the state with our UI.

File: code/web/src/modules/user/MyProducts.js (hunk 8 above)
Comment (tashiad) [reply code/web/src/modules/user/MyProducts.js:37]: These are temporary variables to provide mock data for the purposes of testing this functionality 🙂

File: code/web/src/modules/user/MyAccount.js (hunk 4 above)
Comment (tashiad) [reply code/web/src/modules/user/MyAccount.js:208]: Agreed!

**Review by rachelbuchta — APPROVED** [review]
This all looks great, the styling is really coming together I think. I'm excited to jump in and start refactoring this into redux!!

**Pull request merged**